### PR TITLE
Move to LMDB

### PR DESCRIPTION
**Pull request by Warchant**
Since we urgently need some MVP version of ametsuchi, I decided to implement everything on `lmdb`. 

Implemented commands:
- [x] AssetCreate
- [x] AssetAdd
- [x] AssetRemove
- [x] AssetTransfer
- [x] AccountAdd
- [x] AccountRemove
- [x] PeerAdd
- [x] PeerRemove

What queries will be supported:
- [x] Queries to tx_store:
   - [x] all Asset transactions by public key of creator 
     - [x] for transfer transaction two queries: by public key of receiver and sender
   - [x] all Account transactions by public key of creator
   - [x] all Peer transactions by public key of creator
- [x] Query amount of specific assets (`ledger+domain+asset names`) by account's public key
- [x] Query all assets by account's public key

- [ ] Write tests

Also it supports:
- [x] Autogenerated schema (`*.h` files) from fbs files during compile stage
- [x] All objects are stored as flatbuffers
- [x] Circle 2.0 (for some reason it disappeared)
- [x] Merkle tree
   - [x] push hashes to tree while it is in RAM
   - [x] dump tree to file after commit
   - [x] read tree from file in `init()`
   - [ ] if file with merkle tree does not exist and tx_store contains records, iterate over `tx_store` and calculate new merkle root

### CODE REVIEW

**Comment by grimadas** [comment]
Writing queries to tx_store

File: src/ametsuchi/merkle_tree/merkle_tree.cc [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    i_root_ = leafs_ - 1;          // change root pointer
+    i_current_ = leafs_;           // change pointer to current free cell
+  }
+
+  // copy hash to current empty position
+  tree_[i_current_] = item;
+
+  // find LCA(leftmost leaf, i_current_)
+  // LCA is this many levels above:
+  size_t np = 1 + log2(i_current_ - (leafs_ - 1));
```
Comment (satellitex) [inline src/ametsuchi/merkle_tree/merkle_tree.cc]: I suggest `i_current` is always `leafs_`.
Is it OK?

File: test/generator/tx_generator_test.cc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "tx_generator.h"
+#include <gtest/gtest.h>
+
+TEST(Generator, RandomNumber) {
+  for (size_t i = 0; i < 1000; i++) {
```
Comment (takemiyamakoto) [inline test/generator/tx_generator_test.cc]: ++i

Comment (Warchant) [inline test/generator/tx_generator_test.cc]: it does not matter for simple types :)

Also, it is unfinished generator.

Comment (Warchant) [inline test/generator/tx_generator_test.cc]: WIP

File: src/ametsuchi/merkle_tree/merkle_tree.cc (hunk 1 above)
Comment (Warchant) [inline src/ametsuchi/merkle_tree/merkle_tree.cc]: `i_current_` is a pointer to current "free" leaf

Comment (Warchant) [inline src/ametsuchi/merkle_tree/merkle_tree.cc]: Also, yes, I will add a rollback feature soon. I have to complete generator of transactions.

**Comment by Warchant** [comment]
Approve somebody!

**Comment by grimadas** [comment]
> if file with merkle tree does not exist and tx_store contains records, iterate over tx_store and calculate new merkle root

Is it possible? I think, if merkle_tree is corrupted then tx_store is corrupted too.

**Comment by Warchant** [comment]
Now it is not possible, but in future, it can be.

**Pull request merged**